lib/reek/smells/smell_detector.rb

```diff
@@ -38,10 +39,13 @@ def contexts
       end
 
       def run_for(context)
-        return [] unless enabled_for?(context)
         return [] if exception?(context)
 
-        sniff(context)
+        if enabled_for?(context)
+          sniff(context)
+        else
+          ::Reek::Smells::UnneededSuppression.new(context, self).sniff
```

Comment:
`context` is a parameter to `sniff`, so that won't work as-is. Also, we would be creating a lot of unneeded UnneededSuppression objects.